packages/in_app_purchase/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -203,6 +214,28 @@ private void launchBillingFlow(
       return;
     }
 
+    if (oldSku != null) {
+      SkuDetails oldSkuDetails = cachedSkus.get(oldSku);
+      if (oldSkuDetails == null) {
+        result.error(
+            "NOT_FOUND",
+            "Details for old sku " + sku + " are not available. Has this ID already been fetched?",
+            null);
+        return;
+      }
+    }
+
+    final int prorationMode = getProrationMode(prorationModeValue);
+    if (prorationMode != ProrationMode.UNKNOWN_SUBSCRIPTION_UPGRADE_DOWNGRADE_POLICY) {
+      if (oldSku == null) {
+        result.error(
+            "NOT_FOUND",
+            "oldSku is not available. You must provide the oldSku inorder to use a proration mode.",
+            null);
+        return;
+      }
+    }
```

Comment:
Some idea about this whole error handling section:
```
final int prorationMode = getProrationMode(prorationModeValue);
if (oldSku == null && prorationMode !=ProrationMode.UNKNOWN_SUBSCRIPTION_UPGRADE_DOWNGRADE_POLICY) {
  result.error(
            "IN_APP_PURCHASE_REQUIRE_OLD_SKU",
            "launchBillingFlow  failed because oldSku is null. You must provide a valid oldSku in order to use a proration mode. ",
            null);
        return;
} else if (oldSku != null && !cachedSkus.containsKey(oldSku)) {
  result.error(
            "IN_APP_PURCHASE_INVALID_OLD_SKU",
            "The oldSku provided is invalid in launchBillingFlow. It might because skus were not fetched prior to the call. Please fetch the skus first. An example of how to fetch the skus could be found here: ...",
}
```

And we should add an example of fetching the skus first, then calling launch billing flow with old sku in README, and add a link to the last error message there.


```

Reply:
Done